aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -156,6 +158,29 @@ internal class RealAWSCognitoAuthPlugin(
 
     fun escapeHatch() = authEnvironment.cognitoAuthService
 
+    @Throws(AmplifyException::class)
+    fun initialize() {
+        var token: StateChangeListenerToken? = null
+        val latch = CountDownLatch(1)
+        token = authStateMachine.listen(
+            { authState ->
+                if (authState is AuthState.Configured) {
+                    token?.let(authStateMachine::cancel)
+                    latch.countDown()
+                }
+            },
+            { }
+        )
+        try {
+            latch.await(10, TimeUnit.SECONDS)
+        } catch (e: Exception) {
+            throw AmplifyException(
+                "Failed to configure auth state machine",
+                "Try re-run the application"
```

Comment:
I don't think suggesting re-running the application is a recoverySuggestion, if we keep, it should say "re-running". Alternatively, it would probably be best to suggest checking amplifyconfiguration.json for errors.